caproto/ioc_examples/thread_client_monitor.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+import caproto as ca
+
+import caproto.threading.client
+from caproto.server import pvproperty, PVGroup, ioc_arg_parser, run
+
+from textwrap import dedent
+
+
+_default_thread_context = None
+
+
+def get_default_thread_context():
+    """Get a shared caproto threading client context."""
+    global _default_thread_context
+    if _default_thread_context is None:
+        _default_thread_context = caproto.threading.client.Context()
+    return _default_thread_context
+
+
+def _monitor_pvs(*pv_names, context, queue, data_type='time'):
+    """
+    Monitor pv_names in the given threading context, putting events to `queue`.
+
+    Parameters
+    ----------
+    *pv_names : str
+        PV names to monitor.
+
+    context : caproto.threading.client.Context
+        The threading context to use.
+
+    queue : ThreadsafeQueue
+        Thread-safe queue for the current server async library.
+
+    data_type : {'time', 'control', 'native'}
+        The subscription type.
+
+    Returns
+    -------
+    subscriptions : list
+        List of subscription tuples, with each being:
+        ``(sub, subscription_token, *callback_references)``
+    """
+
+    def add_to_queue(sub, event_add_response):
+        queue.put(('subscription', sub.pv, event_add_response))
```

Comment:
Might as well expose `sub` here, maybe. Not used downstream in this case, but it could be in future variations.

Reply:
Addressed in d625a2f